views/js/controller/SystemStatus/index.js

```diff
@@ -1,39 +1,206 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ *
+ */
 define([
-    'tpl!taoSystemStatus/controller/SystemStatus/reportDetailsModal',
-    'taoSystemStatus/tasksMonitoring/tasksStatistics'
-], function(layout, tasksStatisticsFactory) {
-    'use strict';
-
-    var $container;
-
-    function showTaskReport($modalContent) {
-        var $modal = $(layout({data: $modalContent.html()}));
-        $container.append($modal);
-        $modal.modal({
-            startClosed : true,
-            minWidth : 450
-        });
-        $modal.modal('open');
-
-        $modal.on('closed.modal', function() {
-            $modal.modal('destroy');
-            $(this).remove();
-        });
-    }
-
-    return {
-        start : function() {
-            var tasksStatistics;
-
-            $container = $('#system-status-report');
-            $container.on('click', '.js-feedback-details-button', function () {
-                var $modalContent = $(this).closest('.js-report').find('.js-feedback-details');
-                showTaskReport($modalContent);
-            });
-
-            tasksStatistics = tasksStatisticsFactory();
-            tasksStatistics.render();
+    'jquery',
+    'i18n',
+    'taoSystemStatus/dataProvider/reportsDataProvider',
+    'tpl!taoSystemStatus/controller/SystemStatus/tpl/reportTable',
+    'tpl!taoSystemStatus/controller/SystemStatus/tpl/chartsContainer',
+    'tpl!taoSystemStatus/controller/SystemStatus/tpl/taskQueue',
+    'taoSystemStatus/tasksMonitoring/tasksStatistics',
+    'layout/loading-bar',
+    'ui/feedback',
+    'c3',
+    'ui/modal',
+], function (
+    $,
+    __,
+    reportsDataProvider,
+    reportTableTpl,
+    chartsContainerTpl,
+    taskQueueTpl,
+    tasksStatistics,
+    loadingBar,
+    feedback,
+    c3
+) {
+        'use strict';
 
+        return {
+            start: function () {
+                const $container = $('#system-status-report');
+                const $configurationTablesContainer = $('#system-status-configuration-tables');
+
+                loadingBar.start();
+
+                reportsDataProvider.getReports()
+                    .then((reports) => {
+                        const {
+                            configuration,
+                            configurationValues,
+                            healthCheck,
+                            taskQueueFails,
+                            taskQueueFinished,
+                            taskQueueMonitoring,
+                            redisFreeSpace,
```

Comment:
Those reports may not be provided.
It's better to not rely on existence of particular report.